lib/permissions.ts

```diff
@@ -60,3 +60,27 @@ export const canManageCases = (
 
   return false;
 };
+
+export const canUserAllocateWorkerToPerson = (
+  user: User,
+  person: Resident
+): boolean => {
+  if (user.hasAdminPermissions || user.hasDevPermissions) {
+    return true;
+  }
+
+  if (user.hasChildrenPermissions && person.contextFlag === 'C') {
+    // Children's doesn't require `hasAllocationsPermissions`, as anyone can allocate in CFS
+    return true;
+  }
+
+  if (
+    user.hasAdultPermissions &&
+    person.contextFlag === 'A' &&
+    (user.hasAllocationsPermissions || false)
```

Comment:
Why do we do an OR operation here: `(user.hasAllocationsPermissions || false)`?

Wouldn't `user.hasAllocationsPermissions` be either `true` or `false`, and then when compared to `false`, the OR operation would return the value of `user.hasAllocationsPermissions`. Is this right? What does `|| false` help us to do or check for? 🤔

Reply:
Entertainingly the type def for `user.hasAllocationPermissions` is:

```ts
  hasAllocationsPermissions?: boolean;
```

meaning the potential values here are `true`, `false`, and `undefined`.

The `|| false` I've added effectively defaults it to false if it isn't set.

With that in mind, I actually think the type definition is wrong. I've created this ticket – https://hackney.atlassian.net/browse/SCT-233 – to refactor this type across the app.